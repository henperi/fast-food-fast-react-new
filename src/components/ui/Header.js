import React from "react";
import { NavLink, Link } from "react-router-dom";

const Header = () => (
  <header>
    <nav className="topnav" id="myTopnav">
      <div className="left-nav">
        <Link to="/" className="brand">
          Fast Food Fast
        </Link>
      </div>
      <div className="right-nav">
        <NavLink to="/users/foods" className="">
          Foods Menu
        </NavLink>
        <NavLink to="#about">About Us</NavLink>
        <NavLink to="#contact">Contact Us</NavLink>
        <NavLink to="/signin" className="">
          Signin
        </NavLink>
        <NavLink to="/signup">Signup</NavLink>
        <NavLink className="btn btn-green" to="/users/my-cart">
          My Cart
          <i className="fa fa-shopping-cart" />{" "}
          <span className="count cart-count">0</span>
        </NavLink>
        <a href="javascript:void(0);" className="icon">
          &#9776;
        </a>
      </div>
    </nav>
  </header>
);

export default Header;
