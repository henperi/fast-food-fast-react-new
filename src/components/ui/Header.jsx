import React, { PureComponent, Fragment } from 'react';
import { NavLink, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logoutUser } from '../../actions/userActions';

class Header extends PureComponent {
  handleLogout = (event) => {
    event.preventDefault();

    const { logoutUser: dispatchLogout, history, user } = this.props;
    dispatchLogout();

    localStorage.removeItem('userToken');
    localStorage.removeItem('userId');
    localStorage.removeItem('fullname');
    localStorage.removeItem('role');
  };

  toggleNav = (event) => {
    event.preventDefault();
  };

  toggleNav = (event) => {
    event.preventDefault();
    return (
      event.target.parentElement.parentElement.classList.contains('topnav')
      && event.target.parentElement.parentElement.classList.toggle('responsive')
    );
  };

  render() {
    const { user, cart } = this.props;
    const cartCount = cart.length;

    const IndexLinks = `${window.origin}/` === window.location.href && (
      <Fragment>
        <NavLink to="#about">About Us</NavLink>
        <NavLink to="#contact">Contact Us</NavLink>
      </Fragment>
    );

    const guestNav = (
      <div className="right-nav">
        <NavLink to="/users/foods" className="">
          Foods Menu
        </NavLink>
        {IndexLinks}

        <NavLink to="/signin" className="">
          Signin
        </NavLink>
        <NavLink to="/signup">Signup</NavLink>
        <NavLink className="btn btn-green" to="/users/my-cart">
          My Cart
          <i className="fa fa-shopping-cart" />
          {' '}
          <span className="count cart-count">{cartCount}</span>
        </NavLink>
        <a
          onClick={(e) => {
            this.toggleNav(e);
          }}
          href="/"
          className="icon"
        >
          &#9776;
        </a>
      </div>
    );
    const userNav = (
      <div className="right-nav">
        <NavLink to="/users/foods" activeClassName="active">
          Foods Menu
        </NavLink>
        {IndexLinks}
        <div className="dropdown">
          <NavLink to="/users/my-orders" activeClassName="active">
            My Orders
          </NavLink>

          {/*
            <div className="dropdown-content">
            <NavLink to="/pending-orders">Pending Orders</NavLink>
            <NavLink to="/completed-orders">Completed Orders</NavLink>
            <NavLink to="/all-orders">All My Orders</NavLink>
            </div>
          */}
        </div>
        {/*        <NavLink to="/my-profile" className="">
          My Profile
        </NavLink> */}
        <NavLink className="btn btn-green" to="/users/my-cart">
          My Cart
          <i className="fa fa-shopping-cart" />
          {' '}
          <span className="count cart-count">{cartCount}</span>
        </NavLink>
        <a onClick={this.handleLogout} href="/logout" className="btn-rounded logout">
          Logout
        </a>
        <a
          onClick={(e) => {
            this.toggleNav(e);
          }}
          href="/"
          className="icon"
        >
          &#9776;
        </a>
      </div>
    );
    const adminNav = (
      <div className="right-nav">
        <NavLink to="/admins/foods" className="active">
          Foods Menu
        </NavLink>
        {IndexLinks}
        <div className="dropdown">
          <button type="button" className="dropbtn">
            Orders
            <i className="fa fa-caret-down" />
          </button>
          <div className="dropdown-content">
            <NavLink to="/admins/pending-orders">Pending Orders</NavLink>
            <NavLink to="/admins/completed-orders">Completed Orders</NavLink>
            <NavLink to="/admins/all-orders">All Orders</NavLink>
          </div>
        </div>
        <NavLink to="/admins/my-profile" className="">
          My Profile
        </NavLink>
        <a onClick={this.handleLogout} href="/" className="btn-rounded logout">
          Logout
        </a>
        <a
          onClick={(e) => {
            this.toggleNav(e);
          }}
          href="/"
          className="icon"
        >
          &#9776;
        </a>
      </div>
    );

    let nav = guestNav;
    if (user.isAuthenticated) {
      nav = user.role === 'admin' ? adminNav : userNav;
    }
    return (
      <header>
        <nav className="topnav" id="myTopnav">
          <div className="left-nav">
            <Link to="/" className="brand">
              Fast Food Fast
            </Link>
          </div>
          {nav}
        </nav>
      </header>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user,
  cart: state.cart,
});

export default connect(
  mapStateToProps,
  { logoutUser },
)(Header);
