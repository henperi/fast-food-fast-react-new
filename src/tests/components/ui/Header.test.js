// import ReactShallowRenderer from "react-test-renderer/shallow";
import React from "react";
import Header from "../../../components/ui/Header";
import { shallow } from "enzyme";
// import toJSON from "enzyme-to-json";

test("should render Header correctly", () => {
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();

  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
  // expect((wrapper).find("nav").length).toBe(1);
});
