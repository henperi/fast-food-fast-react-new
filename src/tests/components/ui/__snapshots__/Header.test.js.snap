// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <nav
    className="topnav"
    id="myTopnav"
  >
    <div
      className="left-nav"
    >
      <Link
        className="brand"
        replace={false}
        to="/"
      >
        Fast Food Fast
      </Link>
    </div>
    <div
      className="right-nav"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className=""
        to="/users/foods"
      >
        Foods Menu
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="#about"
      >
        About Us
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="#contact"
      >
        Contact Us
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        className=""
        to="/signin"
      >
        Signin
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/signup"
      >
        Signup
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="btn btn-green"
        to="/users/my-cart"
      >
        My Cart
        <i
          className="fa fa-shopping-cart"
        />
         
        <span
          className="count cart-count"
        >
          0
        </span>
      </NavLink>
      <a
        className="icon"
        href="javascript:void(0);"
      >
        â˜°
      </a>
    </div>
  </nav>
</header>
`;
